pkgs/tools/system/epilys-bb/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, installShellFiles }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "epilys-bb";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "epilys";
+    repo = "bb";
+    rev = "${version}";
+    hash = "sha256-ZIm7JMSZ6HY3Po71rfnRkBzSb9k2d89ldfojffh39+4=";
+  };
+
+  cargoHash = "sha256-lVxdTv5H0AGpLZFi1z5XNIjKbMYhp5k3ipw54YCw62s=";
+
+  meta = with lib; {
+    description = "clean simple fast process viewer";
+    homepage = "https://nessuent.xyz/bb.html";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ cafkafk ];
+  };
```

Comment:
Does not compile/work on `darwin`.
See https://github.com/epilys/bb/issues/3#issuecomment-546596190

```suggestion
    platforms = platforms.linux;
  };
```

and `aarch64-linux` fails due to https://github.com/epilys/bb/commit/c903d4c2975509299fd3d2600a0c4c2102f445d0 missing from v0.4.3

You could either only list `x86_64-linux` in `meta.platform`, essentially preventing this to be built on `aarch64-linux`

```suggestion
    platforms = [ "x86_64-linux" ];
  };
```

Or init at an "unreleased" "unstable" [commit](https://github.com/epilys/bb/commit/c903d4c2975509299fd3d2600a0c4c2102f445d0) instead of v0.4.3:

```patch
diff --git a/pkgs/tools/system/epilys-bb/default.nix b/pkgs/tools/system/epilys-bb/default.nix
index e6e9281d33e..7d490f529c3 100644
--- a/pkgs/tools/system/epilys-bb/default.nix
+++ b/pkgs/tools/system/epilys-bb/default.nix
@@ -1,17 +1,17 @@
 { lib, stdenv, fetchFromGitHub, rustPlatform, installShellFiles }:
 
-rustPlatform.buildRustPackage rec {
+rustPlatform.buildRustPackage {
   pname = "epilys-bb";
-  version = "0.4.3";
+  version = "unstable-2020-12-04";
 
   src = fetchFromGitHub {
     owner = "epilys";
     repo = "bb";
-    rev = "${version}";
-    hash = "sha256-ZIm7JMSZ6HY3Po71rfnRkBzSb9k2d89ldfojffh39+4=";
+    rev = "c903d4c2975509299fd3d2600a0c4c2102f445d0";
+    hash = "sha256-KOXK+1arUWtu/QU7dwXhojIM0faMtwNN3AqVbofq1lY=";
   };
 
-  cargoHash = "sha256-lVxdTv5H0AGpLZFi1z5XNIjKbMYhp5k3ipw54YCw62s=";
+  cargoHash = "sha256-+aCMwKOg+3HDntG14gjJLec8XD51wuTyYyzLjuW6lbY=";
 
   meta = with lib; {
     description = "clean simple fast process viewer";
```

Reply:
Thanks for the thorough review! I went with the latter approach, since it doesn't look like it will be stabilized soon.

I'm unsure if/how I should test that this works?

Reply:
> I'm unsure if/how I should test that this works on `aarch64-linux`?

I've already tested `aarch64-linux` when I wrote that patch :^)

But you can see that ofborg, our CI, was able to built it on `aarch64-linux` as well this time:
https://github.com/NixOS/nixpkgs/pull/235799/checks?check_run_id=13985255641
(or the full logs on https://logs.ofborg.org/?key=nixos/nixpkgs.235799&attempt_id=342f93a6-fa74-442a-9e20-7093e0596b37)

Oh and there is also the `pkgsCross` entrypoint in nixpkgs, which does not require any additional setup on your side and should just work™

```bash
# in your nixpkgs git clone
nix-build -A pkgsCross.aarch64-multiplatform.epilys-bb
```